actions/tag_release.sh

```diff
@@ -37,7 +37,7 @@ git clone ${GIT_REPO} ${LOCAL_REPO}
 
 cd ${LOCAL_REPO}
 echo "Currently at directory `pwd`..."
-git checkout -b ${BRANCH} origin/${BRANCH}
+git checkout -B ${BRANCH} origin/${BRANCH}
```

Comment:
Can you explain the difference of -b and -B?  I looked it up with `git checkout -h` but how is it relevant to this patch?

Reply:
Using the existing `-b`, we see the following error:

```
Cloning into 'st2docker_1508965619_541'...
fatal: A branch named 'master' already exists.
```

This happens because the previous `git clone` points the newly created HEAD to the branch pointed to by the cloned repository's HEAD.. which by default is `"master"` (effectively creating the local `master` branch). We now need to handle the case where `$BRANCH = 'master'` since this script is also used to tag the `st2-docker` repo. According to the man page: If `-B` is given, `<new_branch>` is created if it doesn't exist; otherwise, it is reset. Reset isn't a problem, since we've just cloned into an empty directory (not overwriting anything in the process).

An even better solution might be using `git clone -b ${BRANCH} ...` and getting rid of the `git checkout` invocation altogether. In this PR, I was trying to go for the least number of changes...